### PR TITLE
disttask: mock dispatcher ext in UT

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #48795

Problem Summary:

### What changed and how does it work?
mock dispatcher.Extension for easier testing.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/dispatcher/mock/dispatcher_mock.go
Comment (D3Hunter) [inline pkg/disttask/framework/dispatcher/mock/dispatcher_mock.go:1]: change makefile too

**Review by D3Hunter — APPROVED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Pull request merged**